packages/gatsby/src/internal-plugins/internal-data-bridge/gatsby-node.js

```diff
@@ -112,8 +117,11 @@ exports.sourceNodes = ({
     },
   })
 
+  const gatsbyConfigRootDir = isCompileGatsbyFilesFlagSet()
+    ? `${program.directory}/${COMPILED_CACHE_DIR}`
+    : program.directory
   const pathToGatsbyConfig = systemPath.join(
-    program.directory,
+    gatsbyConfigRootDir,
     `gatsby-config.js`
   )
   watchConfig(pathToGatsbyConfig, createGatsbyConfigNode)
```

Comment:
Is compiled gatsby-config ever updated? In develop .ts file might not get `siteMetadata` hot updates?

Reply:
@pieh At the moment it's only compiled once at the start of the server, not updated

Reply:
For now I think this _can_ be known limitation, but we should think how we would like to address it - do we keep parcel watcher? do we watch config file (we already do above) and instead watch source file and if source is ts file we trigger on-demand compilation for it, etc?